File: DemoMethods/Indexes/TransformerNameAndCountry.cs
using System.Linq;
using Raven.Client.Indexes;

namespace DemoMethods.Indexes
{
    public class TransformerNameAndCountry : AbstractTransformerCreationTask<NameAndCountry.Result>
    {
        public TransformerNameAndCountry()
        {
            TransformResults = results => from result in results
                                          select new
                                          {
                                              result.Name
                                          };
        }
    }

}

File: DemoMethods/src/main/java/net/ravendb/demo/indexes/TransformerNameAndCountry.java
package net.ravendb.demo.indexes;

import net.ravendb.client.indexes.AbstractTransformerCreationTask;

public class TransformerNameAndCountry extends AbstractTransformerCreationTask {

    public TransformerNameAndCountry() {
        transformResults = "from result in results " +
                "select new { " +
                "  Name = result.Name " +
                "}";
    }

}
